lib/internal/modules/esm/resolve.js

```diff
@@ -465,14 +468,15 @@ function isArrayIndex(key) {
  * @param {boolean} internal - Whether the package is internal.
  * @param {boolean} isPathMap - Whether the package is a path map.
  * @param {Set<string>} conditions - The conditions to match.
+ * @param {boolean} mustBeInternalTarget - If the target must be in the package boundary.
```

Comment:
This might be a bit much for a JSDoc comment, but is it possible to briefly explain when we would want this restriction? Is it like `If the target must be in the package scope: yes for "exports", no for "imports"`?